### PR TITLE
RELATED: RAIL-4026 Make user permission related selectors public

### PR DESCRIPTION
**Pull request by plavemar**
- all selectors annotated as public
- docs on permission interface as well as on selectors

JIRA: RAIL-4026

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7086/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8939/ : FAILURE in 2m 01s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7087/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8940/ : SUCCESS in 20m 51s

File: libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts [hunk 1]
```diff
@@ -41,44 +41,143 @@ export const selectCanManageWorkspace = createSelector(selectPermissions, (state
...
+export const selectCanManageMetric = createSelector(selectPermissions, (state) => {
+    return state?.canManageMetric ?? false;
+});
+
+/**
+ * Returns whether the current user has permissions to modify and delete a report object, change report visibility via the `unlisted` flag.
+ *
+ * @public
+ */
+export const selectCanManageReport = createSelector(selectPermissions, (state) => {
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts:154]: This is PixelPerfect-specific it seems, so it is not usable for the Dashboard component imo.

File: libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts [hunk 2]
```diff
@@ -41,44 +41,143 @@ export const selectCanManageWorkspace = createSelector(selectPermissions, (state
...
+export const selectCanExecuteRaw = createSelector(selectPermissions, (state) => {
+    return state?.canExecuteRaw ?? false;
+});
+
+/**
+ * Returns whether the current user has permissions to access workbench.
+ *
+ * @public
+ */
+export const selectCanAccessWorkbench = createSelector(selectPermissions, (state) => {
```
[the remark this answers is not in the document]
Comment (plavemar) [reply libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts:118]: removed

File: libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts [hunk 3]
```diff
@@ -41,44 +41,143 @@ export const selectCanManageWorkspace = createSelector(selectPermissions, (state
...
+    return state?.canAccessWorkbench ?? false;
+});
+
+/**
+ * Returns whether the current user has permissions to create a report object via API.
+ *
+ * @public
+ */
+export const selectCanCreateReport = createSelector(selectPermissions, (state) => {
```
[the remark this answers is not in the document]
Comment (plavemar) [reply libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts:127]: removed

File: libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts [hunk 4]
```diff
@@ -41,44 +41,143 @@ export const selectCanManageWorkspace = createSelector(selectPermissions, (state
 /**
  * Returns whether the current user has permissions to export the report.
```
[the remark this answers is not in the document]
Comment (plavemar) [reply libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts:42]: docs updated

File: libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts (hunk 1 above)
Comment (plavemar) [reply libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts:154]: removed

File: libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts [hunk 5]
```diff
@@ -41,44 +41,143 @@ export const selectCanManageWorkspace = createSelector(selectPermissions, (state
...
+ * Returns whether the current user has permissions to upload CSV files via CSV Uploader.
+ *
+ * @public
+ */
+export const selectCanUploadNonProductionCSV = createSelector(selectPermissions, (state) => {
+    return state?.canUploadNonProductionCSV ?? false;
+});
+
+/**
+ * Returns whether the current user has permissions to download a complete report.
```
[the remark this answers is not in the document]
Comment (plavemar) [reply libs/sdk-ui-dashboard/src/model/store/permissions/permissionsSelectors.ts:105]: docs updated

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2303

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2054/ : SUCCESS in 1m 12s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2337/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1151/ : SUCCESS in 15m 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8956/ : FAILURE in 5m 07s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2303

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2055/ : SUCCESS in 1m 37s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2338/ : SUCCESS in 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1153/ : SUCCESS in 14m 21s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8961/ : SUCCESS in 19m 42s

**Pull request merged**